src/components/DonutChart.js

```diff
@@ -0,0 +1,234 @@
+const React = require('react');
+const d3 = require('d3');
+const objectAssign = require('object-assign');
+
+const DonutPath = require('./DonutPath')
+
+const DonutChart = React.createClass({
+  propTypes: {
+    activeIndex: React.PropTypes.number,
+    activeOffset: React.PropTypes.number,
+    animateOnHover: React.PropTypes.bool,
+    animateOnLoad: React.PropTypes.bool,
+    arcWidth: React.PropTypes.number,
+    baseArcColor: React.PropTypes.string,
+    chartTotal: React.PropTypes.number,
+    children: React.PropTypes.node,
+    colors: React.PropTypes.array,
+    data: React.PropTypes.array.isRequired,
+    dataPoints: React.PropTypes.array,
+    dataPointColors: React.PropTypes.array,
+    dataPointRadius: React.PropTypes.number,
+    height: React.PropTypes.number,
+    labelStyle: React.PropTypes.object,
+    onClick: React.PropTypes.func,
+    onMouseEnter: React.PropTypes.func,
+    onMouseLeave: React.PropTypes.func,
+    opacity: React.PropTypes.number,
+    padAngle: React.PropTypes.number,
+    radiusOffset: React.PropTypes.number,
+    showBaseArc: React.PropTypes.bool,
+    showLegend: React.PropTypes.bool,
+    valueFormat: React.PropTypes.object,
+    valueStyle: React.PropTypes.object,
+    width: React.PropTypes.number,
+    zeroStateText: React.PropTypes.string
+  },
+
+  getDefaultProps () {
+    return {
+      activeIndex: -1,
+      activeOffset: 3,
+      animateOnHover: true,
+      animateOnLoad: true,
+      arcWidth: 80,
+      baseArcColor: '#E5E5E5',
+      colors: d3.scale.category20().range(),
+      data: [],
+      dataPoints: [],
+      dataPointColors: d3.scale.category20b().range(),
+      dataPointRadius: 40,
+      height: 360,
+      labelStyle: {},
+      onClick () {},
+      onMouseEnter () {},
+      onMouseLeave () {},
+      opacity: 1,
+      padAngle: 0.01,
+      radiusOffset: 0,
+      showBaseArc: false,
+      showLegend: true,
+      valueFormat: {},
+      valueStyle: {},
+      width: 360,
+      zeroStateText: 'Roll over an item for details'
```

Comment:
Can you provide a quick explanation of each prop and what the default is what it is? This can be done with an inline comment next to each prop.